Reuse existing PR when available.

greg-1-anderson opened the pull request:
If clu is run at a time when there is already an open pull request with Composer updates that have
not yet been merged, then the existing branch will be updated. This prevents the accumulation of
pull requests should the maintainer be slow about merging updates.

Also, refactored to allow calling clu without any parameters, to operate on an existing local working
copy of a project at the current working directory. This is handy for ad-hoc use, and also for use
in a CI server in circumstances where the site to be updated has already been checked out by the CI
server.

danielbachhuber commented on bin/clu:
```diff
@@ -10,13 +10,27 @@ if ( file_exists( $path = __DIR__ . '/../vendor/autoload.php' )
+  if ( ! $status ) {
+    $repo_url = $first_line_of_output;
+  }
+}
```
Given the rest of the project is already using tabs, could we use tabs instead of spaces for indentation?

greg-1-anderson replied:
Didn't notice that! I will fix it up.

danielbachhuber commented on src/Runner.php:
```diff
@@ -136,6 +139,12 @@ public function start() {
 		}
 
+		if ( $existing_PR_branch ) {
+			// TODO: Add comment to existing PR with $message
```
Do you want to resolve this TODO?

greg-1-anderson replied:
Unfortunately, hub does not have a 'comment' command. I started a pull request (https://github.com/github/hub/pull/1465), but there has been no feedback, so I'm not sure that it's wanted. We could use the curl API.

danielbachhuber commented on src/Runner.php:
```diff
@@ -146,18 +155,17 @@ public function start() {
+		return false;
 	}
-
 }
```
Any ideas on how we could get some basic unit or integration tests going for this?

greg-1-anderson replied:
Functional tests for the basic functionality (pre this PR), it would be pretty easy. We could use `composer create-project` to clone a project that has out-of-date components, and then run `clu` on it. The [Terminus Build Tools plugin](https://github.com/pantheon-systems/terminus-build-tools-plugin) uses similar techniques in its tests.

Functional tests for re-using a PR would be more difficult. Maybe we could cheat a bit to do this step. Create a PR per the above with clu; then, edit the `composer.json` to add a component that was not previously included in the system-under-test. At that point, running `clu` again will behave the same way that it would when there was an update. So, while we would not be testing the same use-case, we would be exercising the code in the same way, and I think that would be sufficient. Creating a mock packagist.org would be too much trouble, and mocking the API calls with something like VCR would also be more work, and less valuable.

I don't think that unit tests for this project would be particularly valuable.

Maybe we could tackle testing as a follow-on task? I have limited time prior to DrupalCon Vienna, but would be happy to help out with this on my return.

greg-1-anderson commented:
Fixed up indentation. To ensure that a project maintains a consistent code style, it is helpful to enforce style rules during the test. I added phpcs + phpcbf rules to enforce TABS instead of SPACES. I did not enforce any other PHP coding standards, though, as the conventions you are following here do not seem to match any of the standards that come bundled with phpcs.  I would suggest adopting a common standard, e.g. PSR-2.

I implemented the commit comment TODO.

I would prefer to put off adding tests until later, per my previous comment.

danielbachhuber commented:
@greg-1-anderson Thanks again. Tagged a v0.2.0 release for this.
